frontend/src/delete-account/DeleteAccount.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+The book project lets a user keep track of different books they would like to read, are currently
+reading, have read or did not finish.
+Copyright (C) 2020  Karan Kumar
+
+This program is free software: you can redistribute it and/or modify it under the terms of the
+GNU General Public License as published by the Free Software Foundation, either version 3 of the
+License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program.
+If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import React from 'react'
+import {Layout} from '../shared/components/Layout';
+import Button from '@material-ui/core/Button';
+import PasswordInput from '../shared/form/Password';
+import {Link} from 'react-router-dom';
+import './DeleteAccount.css'
+
+export default function DeleteAccount():JSX.Element {
```

Comment:
Could we make a call to the delete account endpoint? https://github.com/Project-Books/book-project/blob/0.2.0/backend/src/main/java/com/karankumar/bookproject/backend/controller/UserController.java#L76

- HTTP verb: DELETE
- Takes in the password as a path variable
- Returns 401 unauthorised if the password entered is incorrect

Reply:
Of course, thank you!

Reply:
Hello!
When the user is logged in, there will have to be some sort of cookie, token or header to authentication the user for each request. I was thinking whatever mechanism authenticates the logged in user should be the same mechanism we use for deletion of their is account.
I was reading up on auth practices, and passing any sensitive information in the URL is unsafe. I am hoping you can give me some additional guidance on this.
 I also tried to create an account to see what login auth is doing, but it isn't hooked up.

Reply:
We're using bearer tokens, so we will need to safely store the token and pass it into the header of every request after logging in

Reply:
The endpoint wants me to put the password in the URL. What if we used the tokens to authenticate a delete request instead of passing password as a path variable?

Reply:
We would need to pass the password in. The reason we ask for this is to handle a time-of-check-to-time-of-use issue. (If a user leaves the page open, someone else can try deleting their account. We need to verify that this atomic dangerous action was performed by the intended user. This is similar to how things work in GitHub, Amazon, etc.).

We would still need to pass the JWT token in the header. If it helps, see https://github.com/Project-Books/book-project/wiki/Connecting-to-the-backend-via-Postman.

Reply:
I've updated the PR to include the delete account endpoint. I looked at how the endpoints were hit in Login and used that as a guide.